frontend/src/components/design-system/input/ButtonComponent.vue

```diff
@@ -0,0 +1,267 @@
+<script setup lang="ts">
+import { computed } from 'vue';
+
+import ButtonText from '@/components/design-system/typography/ButtonText.vue';
+
+export interface Props {
+  tag?: 'button' | 'a';
+  href?: string;
+  kind: 'link' | 'outlined' | 'filled' | 'answer';
+  size?: 'medium' | 'small';
+  color?: 'neutral' | 'primary' | 'secondary';
+  selected?: boolean;
+}
+
+const props = withDefaults(defineProps<Props>(), {
+  tag: 'button',
+  href: undefined,
+  size: 'medium',
+  color: 'neutral',
+  selected: false,
+});
+
+const classes = computed(() => ({
+  [`button--${props.kind}`]: props.kind,
+  [`button--${props.size}`]: props.size,
+  [`button--${props.color}`]: props.color,
+  [`button--${props.selected}`]: props.selected,
+}));
+
+const computedSize = computed(() =>
+  props.kind === 'link' ? props.size : 'medium'
+);
+</script>
+
+<template>
+  <component :is="tag" :href="href" :class="['button', classes]">
+    <span
```

Comment:
icon je reseny schvalne jako \<slot/\> a ne prop `iconName` aby clovek mohl teoreticky poslat neco jinyho nez `IconComponent` ?

Reply:
Spíš kvůli tomu tree shakingu, jinak by se tam musel ještě pořešit dynamický import té ikonky. O což jsem se nějak předtím pokoušel a nepovedlo se mi to udělat, teď bych tím neztrácel čas.